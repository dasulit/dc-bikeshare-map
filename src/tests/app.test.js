import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import App from '../components/app';

describe('App', () => {
  test('renders', () => {
    const wrapper = shallow(React.createElement(App));
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test('fetches bike data and sets state', async () => {
    jest.spyOn(global, 'fetch').mockImplementation(() =>
      Promise.resolve({
        json: () =>
          Promise.resolve({
            features: [{ id: 'feature-1' }],
            type: 'FeatureCollection'
          }),
        status: 200
      })
    );
    const wrapper = shallow(React.createElement(App));
    expect(global.fetch).toHaveBeenCalledTimes(1);
    expect(global.fetch).toHaveBeenCalledWith('/bikes');
    setTimeout(() => {
      expect(wrapper.state()).toEqual({
        bikeData: {
          features: [{ id: 'feature-1' }],
          type: 'FeatureCollection'
        }
      });
    });
  });
});
