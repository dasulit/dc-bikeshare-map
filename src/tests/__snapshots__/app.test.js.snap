// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div>
  <MapView
    bikeData={
      Object {
        "features": Array [],
        "type": "FeatureCollection",
      }
    }
  />
  <div
    className="overlay"
  >
    <h1
      className="margin-bottom-10"
    >
      Rentable Bikes in Washington, DC
    </h1>
    <p>
      Explore bikes available to rent from Capital Bikeshare and JUMP.
    </p>
  </div>
</div>
`;
